src/mono/mono.proj

```diff
@@ -461,20 +462,19 @@
     <PropertyGroup Condition="'$(TargetsiOS)' == 'true' or '$(TargetstvOS)' == 'true' or '$(TargetsMacCatalyst)' == 'true'">
       <_IcuLibdir>$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)/runtimes/$(TargetOS)-$(TargetArchitecture)/native/lib</_IcuLibdir>
     </PropertyGroup>
-    <ItemGroup Condition="('$(TargetsOSX)' == 'true' or '$(TargetsMacCatalyst)' == 'true') and '$(Platform)' == 'arm64'">
-      <_MonoCMakeArgs Include="-DCMAKE_OSX_ARCHITECTURES=arm64"/>
-    </ItemGroup>
     <ItemGroup Condition="'$(TargetsiOS)' == 'true' or '$(TargetstvOS)' == 'true'">
       <_MonoCMakeArgs Include="-DCMAKE_SYSTEM_NAME=$(_MonoCMakeSystemName)"/>
       <_MonoCMakeArgs Include="-DCMAKE_OSX_DEPLOYMENT_TARGET=$(_MonoCMakeVersionMin)" />
       <_MonoCMakeArgs Include="-DCMAKE_OSX_SYSROOT='$(_MonoCMakeSysroot)'" />
-      <_MonoCMakeArgs Condition="'$(Platform)' == 'x64'" Include="-DCMAKE_OSX_ARCHITECTURES=x86_64"/>
-      <_MonoCMakeArgs Condition="'$(Platform)' == 'x86'" Include="-DCMAKE_OSX_ARCHITECTURES=i386"/>
-      <_MonoCMakeArgs Condition="'$(Platform)' == 'arm64'" Include="-DCMAKE_OSX_ARCHITECTURES=arm64"/>
-      <_MonoCMakeArgs Condition="'$(Platform)' == 'arm'" Include="&quot;-DCMAKE_OSX_ARCHITECTURES=armv7%3Barmv7s&quot;"/>
       <_MonoCFLAGS Include="-Wl,-application_extension" />
       <_MonoCXXFLAGS Include="-Wl,-application_extension" />
     </ItemGroup>
+    <ItemGroup Condition="$([MSBuild]::IsOSPlatform('OSX'))">
```

Comment:
nit: I'd prefer if this uses TargetsOSX, TargetsMacCatalyst, TargetsiOS, TargetstvOS instead.

Reply:
I do not have a strong opinion. Statiscially, this file in main branch has 9 occurances of `TargetsOSX` and 10 occurances of `[MSBuild]::IsOSPlatform('OSX')`. On the other side of the house, we added these properties: https://github.com/dotnet/runtime/blob/7526a4c6a0a095670f1135d983a349b4e867f2e2/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets#L29-L30 where we want `any darwin based platform`.

IMO, lets give it a big sweep across the repo for `any darwin based platform` by defining `TargetsApplePlatform` (or similar) in top-level `Directory.Build.props` in a separate PR. I can follow-up on this.